### ISSUE TITLE
Error in delete asset

### ISSUE THREAD

**Issue by Farneyaurelio** [Dolibarr/dolibarr#25294]
### Bug

I have a doubt, I’m doing a test with a module under development “Assets”, when I try to delete an asset that I create it gives this error.
I was doing the translation in the Dolibarr translations part, could it have something related to this?

### Environment Version

v17.0.2

### Environment Web server

Linux

### Environment PHP

PHP 7.3

### Environment Database

mysql

### Environment URL(s)

Asset module

### Expected and actual behavior

must delete the selected asset

### Steps to reproduce the behavior

when I selected or click on the "remove" button the error appears:
Cannot delete or update a parent row: a foreign key constraint fails (`lanix103_doli404`.`dolb_asset_depreciation`, CONSTRAINT `fk_asset_depreciation_asset` FOREIGN KEY (`fk_asset`) REFERENCES `dolb_asset` (`rowid`))

### Attached files

![2023-07-05_19h34_53](https://github.com/Dolibarr/dolibarr/assets/116181924/fd733d18-88ed-4d06-b498-ee555bf6ea30)

**Comment by priojk** [comment]
I just checked in the v18-beta after a fresh install. There I could not reproduce the problem. So it's either already fixed or there are some special conditions in your case. Are you using a version which got upgraded from an older installation? Maybe some migration that went wrong?

**Comment by Farneyaurelio** [comment]
Hi Priojk, all migrations went smoothly. As you commented, it may be a problem that has already been solved in later versions then. As I am effectively working in the company with Dolibarr, I cannot use version 18 which is unstable, I will wait for its official release.
Thanks for the feedback.

**Comment by Farneyaurelio** [comment]
> I just checked in the v18-beta after a fresh install. There I could not reproduce the problem. So it's either already fixed or there are some special conditions in your case. Are you using a version which got upgraded from an older installation? Maybe some migration that went wrong?

**Comment by Farneyaurelio** [comment]
Hi Priojk, I performed a diagnosis more calmly, and I realized that this error happens because when we create the asset and specify the resource model it automatically calculates its depreciation. Therefore, this asset has lines in the database mentioned in the error. When I manually delete these rows in the database I can delete the asset. Is there any command within dolibarr to delete this depreciation calculation?

Another question, are you already thinking of an output such as a table or report for this module?
